kythe/typescript/testdata/generic.ts

```diff
@@ -33,6 +33,27 @@ function generic<T>(x: T, y: IFace) {
   return x;
 }
 
+// Simple constrained generic.
+//- @T defines/binding ConstrainedGenericT
+//- @IFace ref IFace
+//- ConstrainedGenericT bounded/upper IFace
+function constrainedGeneric<T extends IFace>() {}
+
+// Constrained generic with reference to type param within it.
+//- @#0T defines/binding ConstrainedGenericT2
+//- !{@#0T ref ConstrainedGenericT}
+//- @#1T ref ConstrainedGenericT2
+function constrainedGenericRef<T, K extends keyof T>() {}
```

Comment:
Are there some assertions on K that could be made here?

Reply:
Added some.